
# Row and column proportions

## Setup

In [1]:
import pandas as pd

## Data

In [2]:
ROOT = "https://raw.githubusercontent.com/kirenz/datasets/master/"
DATA = "loans.csv"

df = pd.read_csv(ROOT + DATA)

df.homeownership = df.homeownership.astype("category")
df.application_type = df.application_type.astype("category")

## Contingency table with proportions

Normalize by dividing all values by the sum of values:

- If passed ‘all’ or True, will normalize over all values.

- If passed ‘index’ will normalize over each row.

- If passed ‘columns’ will normalize over each column.

- If margins is True, will also normalize margin values.

### Row proportions

In [3]:
# A contingency table with row proportions for the application type and homeownership variables.
pd.crosstab(df.application_type	, df.homeownership, normalize='index').round(4)*100

homeownership     mortgage    own   rent
application_type                        
individual           45.14  13.76  41.11
joint                63.55  12.24  24.21

- 0.411 corresponds to the proportion of individual applicants who rent.

### Column proportions

In [4]:
# A contingency table with row proportions for the application type and homeownership variables.
pd.crosstab(df.application_type	, df.homeownership, normalize='columns').round(4)*100

homeownership     mortgage    own   rent
application_type                        
individual           80.16  86.47  90.62
joint                19.84  13.53   9.38

- The value 0.906 indicates that 90.6% of renters applied as individuals for the loan. 
- This rate is higher compared to loans from people with mortgages (80.2%) or who own their home (86.5%)